worlds/yugioh06/rules.py

```diff
@@ -0,0 +1,878 @@
+        "Curse of Vampire",
+        "Vampire Lord",
+        "Goblin Zombie",
+        "Book of Life",
+        "Call of the Mummy"
+    ], player) >= 6
+
+
+def only_dragon(state, player):
+    return state.count_from_list_exclusive([
+        "Luster Dragon",
+        "Spear Dragon",
+        "Cave Dragon"
+    ], player) >= 1 and (state.count_from_list_exclusive([
+        "Luster Dragon",
+        "Spear Dragon",
+        "Cave Dragon"
+        "Armed Dragon LV3",
+        "Masked Dragon",
+        "Twin-Headed Behemoth",
+        "Element Dragon",
+        "Troop Dragon",
+        "Horus the Black Flame Dragon LV4",
+        "Stamping Destruction"
+    ], player) + (state.count_from_list_exclusive([
+        "Luster Dragon #2",
+        "Armed Dragon LV5",
+        "Kaiser Glider",
+        "Horus the Black Flame Dragon LV6"
+    ], player) >= 1)) >= 7
+
+
+def only_spellcaster(state, player):
+    return state.count_from_list_exclusive([
+        "Dark Elf",
+        "Gemini Elf",
+        "Skilled Dark Magician",
+        "Toon Gemini Elf",
+        "Kycoo the Ghost Destroyer",
+        "Familiar-Possessed - Aussa"
+    ], player) >= 1 and (state.count_from_list_exclusive([
+        "Dark Elf",
+        "Gemini Elf",
+        "Skilled Dark Magician",
+        "Toon Gemini Elf",
+        "Kycoo the Ghost Destroyer",
+        "Familiar-Possessed - Aussa",
+        "Breaker the magical Warrior",
+        "The Tricky",
+        "Injection Fairy Lily",
+        "Magician of Faith",
+        "Tsukuyomi",
+        "Gravekeeper's Spy",
+        "Gravekeeper's Guard",
+        "Summon Priest",
+        "Old Vindictive Magician",
+        "Apprentice Magician",
+        "Magical Dimension"
+    ], player) + (state.has_any([
+        "Chaos Command Magician",
+        "Cybernetic Magician"
+    ], player))) >= 7
+
+
+def equip_unions(state, player):
+    return ((state.has("Burning Beast", player) and state.has("Freezing Beast", player) and
+            state.has("Metallizing Parasite - Lunatite", player) and state.has("Mother Grizzly", player)) or
+            (state.has("Dark Blade", player) and state.has("Pitch-Dark Dragon", player) and
+            state.has("Giant Orc", player) and state.has("Second Goblin", player) and
+            state.has("Mystic Tomato", player)) or
+            (state.has("Decayed Commander", player) and state.has("Zombie Tiger", player) and
+            state.has("Vampire Orchis", player) and state.has("Des Dendle", player) and
+            state.has("Giant Rat", player)) or
+            (state.has("Indomitable Fighter Lei Lei", player) and state.has("Protective Soul Ailin", player) and
+            state.has("V-Tiger Jet", player) and state.has("W-Wing Catapult", player) and
+            state.has("Shining Angel", player)) or
+            (state.has("X-Head Cannon", player) and state.has("Y-Dragon Head", player) and
+            state.has("Z-Metal Tank", player) and state.has("Shining Angel", player))) and \
```

Comment:
This seems like it could be a chain of

```
state.has_all([...])
or state.has_all([...])
or state.has_all([...])
...
```

Apologies if I commented on this before.

*Low priority*